### PR TITLE
load server-side key via CLI

### PR DESCRIPTION
**Pull request by bigbrett**
- Adds the ability to load a key into the server keystore with the `--key` and `--id` CLI arguments. Required for wolfBoot simulator support
- Wraps the example server in `while(1)` and reinitializes the server context on client disconnect

### CODE REVIEW

**Comment by bigbrett** [comment]
@billphipps ug, needs your fixes in #15 to get merged first, otherwise will keep failing CI

File: posix/tcp/wh_server_tcp/wh_server_tcp.c [hunk 1]
```diff
@@ -4,128 +4,260 @@
...
     FLASH_RAM_SIZE = 1024 * 1024,
 };
 
 #define WH_SERVER_TCP_IPSTRING "127.0.0.1"
 #define WH_SERVER_TCP_PORT 23456
-#define WH_SERVER_ID 56
+#define WH_SERVER_ID 57
 
-static int wh_ServerTask(void* cf)
+static int loadAndStoreKeys(whServerContext* server, whKeyId* outKeyId,
```
Comment (billphipps) [inline posix/tcp/wh_server_tcp/wh_server_tcp.c:39]:
```suggestion
static int _LoadAndStoreKeys(whServerContext* server, whKeyId* outKeyId,
```
Been trying to make all static functions start with _

File: posix/tcp/wh_server_tcp/wh_server_tcp.c [hunk 2]
```diff
@@ -4,128 +4,260 @@
...
+                            const char* keyFilePath, int keyId, int clientId)
+{
+    int           ret;
+    int           keyFd;
+    int           keySz;
+    char          keyLabel[] = "baby's first key";
```
Comment (billphipps) [inline posix/tcp/wh_server_tcp/wh_server_tcp.c:45]: Maybe consider using basename(keyFilePath) for the label?

File: posix/tcp/wh_server_tcp/wolfhsm_cfg.h [hunk 3]
```diff
@@ -35,7 +34,11 @@
 #define WOLFHSM_CFG_NVM_OBJECT_COUNT 32
 #define WOLFHSM_CFG_SERVER_KEYCACHE_COUNT 10
 #define WOLFHSM_CFG_SERVER_KEYCACHE_SIZE 1024
+#define WOLFHSM_CFG_SERVER_KEYCACHE_BIG_BUFSIZE 4096
+
 #define WOLFHSM_CFG_SERVER_DMAADDR_COUNT 8
 #define WOLFHSM_CFG_SERVER_CUSTOMCB_COUNT 8
 
+#define XMEMFENCE() __atomic_thread_fence(__ATOMIC_SEQ_CST)
```
Comment (billphipps) [inline posix/tcp/wh_server_tcp/wolfhsm_cfg.h:42]: Are you sure you need this override for the example?  I think the autodetect logic should pick the right one.

**Review by billphipps — APPROVED** [review]
All good.  Minor nits.  Feel free to completely ignore.

**Pull request merged**